pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -26,6 +26,21 @@ import (
 	"github.com/openshift/origin/pkg/image/importer"
 )
 
+const (
+	// DockerRegistryURLEnvVar is a mandatory environment variable name specifying external url of internal
+	// docker registry. All references to pushed images will be prefixed with its value.
+	DockerRegistryURLEnvVar = "DOCKER_REGISTRY_URL"
+
+	// EnforceQuotaEnvVar is an boolean environment variable that allows to turn quota enforcement on or off.
+	// By default, quota enforcement is off. Recognized values are "true", and "false".
+	EnforceQuotaEnvVar = "OPENSHIFT_ENFORCE_QUOTA"
+
+	// ProjectCacheTTLEnvVar is an environment variable specifying an eviction timeout for project quota
```

Comment:
That surprises me - I would expect env vars corresponding to options to
override.  Is there a bug or justification for that you can link to?

On Fri, Apr 8, 2016 at 4:03 AM, Michal Minar <notifications@github.com>
wrote:

> In pkg/dockerregistry/server/repositorymiddleware.go
> <https://github.com/openshift/origin/pull/8195#discussion_r58992018>:
>
> > @@ -26,6 +26,21 @@ import (
> >  	"github.com/openshift/origin/pkg/image/importer"
> >  )
> >
> > +const (
> > +	// DockerRegistryURLEnvVar is a mandatory environment variable name specifying external url of internal
> > +	// docker registry. All references to pushed images will be prefixed with its value.
> > +	DockerRegistryURLEnvVar = "DOCKER_REGISTRY_URL"
> > +
> > +	// EnforceQuotaEnvVar is an boolean environment variable that allows to turn quota enforcement on or off.
> > +	// By default, quota enforcement is off. Recognized values are "true", and "false".
> > +	EnforceQuotaEnvVar = "OPENSHIFT_ENFORCE_QUOTA"
> > +
> > +	// ProjectCacheTTLEnvVar is an environment variable specifying an eviction timeout for project quota
>
> Unfortunately, middleware options cannot be overriden by environment
> variables. We can support configuration via config file for sure, but we
> need to allow for easy overrides as well unless we start generating a
> config file as a secret during oadm registry.
>
> I'll allow for configuration with the config file but still allow for
> overrides with these variables.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8195/files/182833487b48f39d14835bb857f60d2f9aad8099#r58992018>
>